### PR TITLE
chore: cache docker builds in github actions

### PR DESCRIPTION
**Pull request by mscolnick**
This uses https://docs.docker.com/build/cache/backends/gha/ to cache the docker builds via in GitHub actions

### CODE REVIEW

File: .github/workflows/seed-dockers.yml [hunk 1]
```diff
@@ -22,12 +25,12 @@ jobs:
...
-            ts: 
+            ts:
               - 'docker/seed/Dockerfile.ts'
-              - '.github/workflows/build-seed.yml'
-            java: 
+              - '.github/workflows/seed-dockers.yml'
+            java:
               - 'docker/seed/Dockerfile.java'
-              - '.github/workflows/build-seed.yml'
+              - '.github/workflows/seed-dockers.yml'
```
Comment (mscolnick) [inline .github/workflows/seed-dockers.yml:33]: i think there was a file namechange that was missed

File: .github/workflows/publish-generator.yml [hunk 2]
```diff
@@ -130,6 +133,8 @@ jobs:
           context: ./generators/python
           file: ./generators/python/pydantic/Dockerfile
           platforms: linux/amd64,linux/arm64
+          cache-from: type=gha
```
Comment (armandobelardo) [inline .github/workflows/publish-generator.yml:136]: Meganit: do we want to specify a cache-from here?

Also should we do this for every docker push step?

**Review by armandobelardo — APPROVED** [review]
Approving, just the one comment on doing this for every generator container!

File: .github/workflows/publish-generator.yml (hunk 2 above)
Comment (mscolnick) [reply .github/workflows/publish-generator.yml:136]: oh woops didnt realize there were so many more of them. ill add them below too.

> Meganit: do we want to specify a cache-from here?
are you suggesting putting it somewhere else? i think this is the best spot for it.

**Review by armandobelardo — APPROVED** [review]

**Pull request merged**